qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -439,6 +439,7 @@ class WebEngineTab(browsertab.AbstractTab):
     """A QtWebEngine tab in the browser."""
 
     WIDGET_CLASS = QOpenGLWidget
+    history_prepared = []
```

Comment:
If you assign this here on class level, every `WebEngineTab` will share the same `history_prepared` data - I think you should do `self.history_prepared = []` in `__init__`.

You should also do this in `browsertab.AbstractTab` instead, not in each tab subclass individually.